unattended-upgrade

```diff
@@ -453,6 +456,38 @@ class Unlocked:
             apt_pkg.pkgsystem_lock()
 
 
+class OriginPkgs(defaultdict):
+    """
+    Package names by highest allowed pretty-printed origin
+
+    """
+    def add(self, pkg, cache):
+        # type: (apt.Package, UnattendedUpgradesCache) -> None
+        allowed_origins = cache.allowed_origins
+        for origin in ver_in_allowed_origin(pkg, allowed_origins).origins:
+            if is_allowed_origin(origin, allowed_origins):
+                self[origin.origin + " " + origin.archive].add(pkg.name)
+                return
+
+    def pop_upgradable(self, cache):
```

Comment:
This is confusing, why is that needed? 

Answer: Because we only want to list kept-back packages; but this is not explicit in the name of the class (or method?)

Reply:
How about naming the class KeptPackages and adding some comments? Do you have a better idea for the name?

Reply:
@julian-klode I tried to clear the confusion, do you think it is good now?

Reply:
Yeah